torch/_dynamo/symbolic_convert.py

```diff
@@ -520,12 +520,21 @@ class InstructionTranslatorBase(Checkpointable[InstructionTranslatorGraphState])
     accept_prefix_inst: bool
     prefix_insts: List[Instruction]
     inline_depth: int
+    inconsistent_side_effects: bool
```

Comment:
Please propagate this from inlining translator to parent

Reply:
Also please add a test showing the inconsistent history within user function call, and generator

Reply:
You may do so here:
https://github.com/pytorch/pytorch/blob/217523b4366bf6a8f2e7808a24d276ed5242d710/torch/_dynamo/symbolic_convert.py#L2383

Reply:
Technically, the bug would not be caught here - if we decide to skip, then we'll try to compile the user function on its own.

This is a fairly interesting decision - one would think that if we decide to skip the parent scope, we'd skip the child scope as well.

I think we could improve compile time when there are a lot of nested, non-tensor-effectful functions. - becaused we've already traced all calls to such functions...

Reply:
Actually, we already skip majority of such cases:
https://github.com/pytorch/pytorch/blob/217523b4366bf6a8f2e7808a24d276ed5242d710/torch/_dynamo/convert_frame.py#L218

Reply:
We do at least run into the error for generators since dynamo won't convert them as a top-level construct.

Reply:
There is actually a bit of ambiguity about whether one would really like to trigger compilation for the parent frame. In a way, triggering compilation only for the user function in question is good in the sense that it localizes the damage. On the other hand, I believe it would result in unnecessary graph break.